server/service/src/invoice_line/stock_out_line/update/validate.rs

```diff
@@ -63,6 +65,46 @@ pub fn validate(
     check_location_on_hold(&batch_pair.main_batch).map_err(|e| match e {
         LocationIsOnHoldError::LocationIsOnHold => LocationIsOnHold,
     })?;
+
+    // If we have an allocated_date older than 24hours, we need to calculate the historical stock line to see if we would have enough stock at that time
```

Comment:
Is this safe enough? Figure if it's the same day, then we assume it's not backdated?